core/unit/name_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package unit
+
+import (
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+type unitNameSuite struct{}
+
+var _ = gc.Suite(&unitNameSuite{})
+
+func (*unitNameSuite) TestNameValidate(c *gc.C) {
+	tests := []struct {
+		name string
+		err  error
+	}{
+		{
+			name: "",
+			err:  InvalidUnitName,
+		},
+		{
+			name: "invalid",
+			err:  InvalidUnitName,
+		},
+		{
+			name: "valid/0",
+		},
+		{
+			name: "valid/-1",
+			err:  InvalidUnitName,
+		},
+		{
+			name: "valid/10000",
+		},
```

Comment:
**suggestion:** Please be more creative and add few more valid and invalid names:

* invalid/bozo-6o-76
* valid/bozo-6o-z6

This kind of stuff. Being creative will ensure that we don't miss a case.

Reply:
Added a bunch more cases